test/gtest/common/mem_buffer.cc

```diff
@@ -229,11 +229,20 @@ uint64_t mem_buffer::pat(uint64_t prev) {
 }
 
 mem_buffer::mem_buffer(size_t size, ucs_memory_type_t mem_type) :
-    m_mem_type(mem_type), m_ptr(allocate(size, mem_type)), m_size(size) {
+    m_mem_type(mem_type), m_ptr(allocate(size, mem_type)), m_size(size),
+    m_allocated(true) {
+}
+
+mem_buffer::mem_buffer(void *ptr, size_t size, ucs_memory_type_t mem_type) :
+    m_mem_type(mem_type), m_ptr((ptr != NULL) ? ptr : allocate(size, mem_type)),
+    m_size(size), m_allocated(ptr == NULL)
```

Comment:
safer to initialize m_ptr(ptr) and m_allocated(false) here call allocate() from body (if m_ptr == NULL) and set m_allocated = true inside 
then destructor would work correctly in case of allocation failure

Reply:
don't see issue here, but ok, fixed